src/JuliusSweetland.OptiKey.Core/UI/ViewModels/MainViewModel.ServiceEventHandlers.cs

```diff
@@ -2279,15 +2280,66 @@ private void HandleFunctionKeySelectionResult(KeyValue singleKeyValue)
                     break;
 
                 case FunctionKeys.Speak:
+                    string textToSpeak = "";
+                    if (!string.IsNullOrEmpty(keyboardOutputService.Text))
+                    {
+                        if (keyboardOutputService.Text.Contains(TranslationService.LICENSE_TEXT))
+                        {
+                            textToSpeak = keyboardOutputService.Text.Replace(TranslationService.LICENSE_TEXT, "");
+                        }
+                        else
+                        {
+                            textToSpeak = keyboardOutputService.Text;
+                        }
+                    }
+
                     var speechStarted = audioService.SpeakNewOrInterruptCurrentSpeech(
-                        keyboardOutputService.Text,
+                        textToSpeak,
                         () => { KeyStateService.KeyDownStates[KeyValues.SpeakKey].Value = KeyDownStates.Up; },
                         Settings.Default.SpeechVolume,
                         Settings.Default.SpeechRate,
                         Settings.Default.SpeechVoice);
                     KeyStateService.KeyDownStates[KeyValues.SpeakKey].Value = speechStarted ? KeyDownStates.Down : KeyDownStates.Up;
                     break;
 
+                case FunctionKeys.Translation:
+                    {
+                        Log.Info("Translating text from scratchpad.");
+                        string textFromScratchpad = KeyboardOutputService.Text;
+
+                        if (!string.IsNullOrEmpty(textFromScratchpad))
+                        { 
+                            TranslationService.Response response = await translationService.translate(textFromScratchpad);
+                            if (response.status == "Error")
+                            {
+                                string errorToShow;
+
+                                if (response.exceptionMessage == "") {

```

Comment:
Could you please push the curly bracket onto the next line. Extremely minor I know...